vispy/scene/canvas.py

```diff
@@ -241,73 +262,205 @@ def draw_visual(self, visual, event=None, viewport=None):
         event : None or DrawEvent
             Optionally specifies the original canvas draw event that initiated
             this draw.
-        viewport : tuple | None
-            Optionally specifies the viewport to use. If None, the entire
-            physical size is used.
         """
-        self.set_current()
         prof = Profiler()
```

Comment:
Is this profiler stuff supposed to be here, or is it an artifact that should be removed?